packages/devtools_app/lib/src/ui/flutter/service_extension_widgets.dart

```diff
@@ -61,16 +62,15 @@ class _ServiceExtensionButtonGroupState
       // VMServiceManager.
       final extensionName = extension.description.extension;
       // Update the button state to match the latest state on the VM.
-      _subscriptions.add(serviceManager.serviceExtensionManager
+      autoDispose(serviceManager.serviceExtensionManager
           .getServiceExtensionState(extensionName, (state) {
         setState(() {
           extension.isSelected =
               state.value == extension.description.enabledValue;
         });
       }));
       // Track whether the extension is actually exposed by the VM.
-      _subscriptions
-          .add(serviceManager.serviceExtensionManager.hasServiceExtension(
+      autoDispose(serviceManager.serviceExtensionManager.hasServiceExtension(
```

Comment:
why not put `autoDispose` in the service_manager methods directly, that way any caller of `hasServiceExtension` or `getServiceExtensionState` can reuse the functionality.

Reply:
Ah that makes sense.